### ISSUE TITLE
Parser bug

### ISSUE THREAD

**Issue by b2renger** [AtelierNum/AtelierNum.github.io#16]
Dans les repo workshop code créatif 1819 et 1920 les liens vers les cahiers de recherches ne renvoient pas vers la consultation des pdf sur github.

Dans 1819 => 404 : avec une adresse mal constuire
Dans 1929 => il ne se passe rien

**Comment by AtelierNum** [comment]
Le lien vers le portfolio de Raphaël Perraud dans la rubrique about souffre aussi d'un bug de parser

**Comment by b2renger** [comment]
https://ateliernum.github.io/#/courses/p5js-designing-interactive-patterns
ne s'affiche pas, peut-être un souci de caractères ?

**Comment by R4ph3rd** [comment]
> Le lien vers le portfolio de Raphaël Perraud dans la rubrique about souffre aussi d'un bug de parser
Non, j'avais simplement mal addressé l'url : je n'avais pas ajouté le header http dans l'url, donc ça renvoyait vers une page locale

**Comment by R4ph3rd** [comment]
J'ai corrigé pour les pdf également. J'ai voulu faire en sorte que ça l'ouvre dans un nouvel onglet en ajoutant un target blank, mais l'opération était vraiment lourde (tout du moins avec la méthode que j'ai trouvée) donc je l'ai retiré. J'y reviendrai peut etre plus tard quand j'aurais une meilleure solution.
Pour l'instant j'aligne les conditions qui vérifient si l'url en cours de mixage contient l'une ou l'autre bout de string, et c'est en plein dans le code, donc pas forcément lisible. Peut être qu'à l'avenir tu auras besoin d'ajouter plus d'extensions à éviter de parser dans les url, donc ce serait plus pratique d'instaurer une liste dans **data**, que tu pourrais facilement éditer.
Qu'en penses-tu @b2renger ?

**Comment by b2renger** [comment]
Pourquoi pas, mais ne suis pas sûr que cela soit nécessaire. Dans l'absolu télécharger des zips, mais ça je peux le faire en uploadant dans la section release de github et filer le lien qui download directement.

A part player vidéo / audio / et pdf, je vois pas trop de quoi d'autre on pourrait avoir besoin

**Comment by R4ph3rd** [comment]
Ah oui audio ; il n'est pas pris en compte pour le moment ; ça devrait renvoyer vers quoi du coup ? github ou on ouvre le fichier en local ?
L'audio il peut y avoir plusieurs formats à checker du coup, à moins que tu les enregistre tous sous le même format ?

**Comment by b2renger** [comment]
Hum pour l'audio je me dis de iframe comme pour les vidéos. Je pense pas qu'on va heberger des fichiers purement audio. Donc je pense qu'il faut se simplifier la vie. Si audio il y a ça sera posté sur soundcloud ou autre et on embed un player. Non ?

**Comment by R4ph3rd** [comment]
Oui effectivement c'est mieux, ça emportera toutes les features de gestion de l'audio directement, ce sera plus simple. À tester pour voir si c'est déjà pris en compte dans le parser !
